app/code/Magento/Ui/view/base/web/js/grid/masonry.js

```diff
@@ -0,0 +1,261 @@
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+define([
+    './listing',
+    'jquery',
+    'ko'
+], function (Listing, $, ko) {
+    'use strict';
+
+    return Listing.extend({
+        defaults: {
+            template: 'Magento_Ui/grid/masonry',
+            imports: {
+                rows: '${ $.provider }:data.items',
+                errorMessage: '${ $.provider }:data.errorMessage'
+            },
+            listens: {
+                'rows': 'initComponent'
+            },
+
+            /**
+             * Images container id
+             * @param string
+             */
+            containerId: null,
+
+            /**
+             * Minimum aspect ratio for each image
+             * @param int
+             */
+            minRatio: null,
+
+            /**
+             * Container width
+             * @param int
+             */
+            containerWidth: window.innerWidth,
+
+            /**
+             * Margin between images
+             * @param int
+             */
+            imageMargin: 20,
+
+            /**
+             * Maximum image height value
+             * @param int
+             */
+            maxImageHeight: 240
+        },
+
+        /**
+         * Init observable variables
+         * @return {Object}
+         */
+        initObservable: function () {
+            this._super()
+                .observe([
+                    'rows',
+                    'errorMessage'
+                ]);
+
+            return this;
+        },
+
+        /**
+         * Init component handler
+         * @param {Object} rows
+         * @return {Object}
+         */
+        initComponent: function (rows) {
+            if (!rows || !rows.length) {
+                return;
+            }
+            this.imageMargin = parseInt(this.imageMargin, 10);
+            this.container = $('[data-id="' + this.containerId + '"]')[0];
+
+            this.setLayoutStyles();
+            this.setEventListener();
+
+            return this;
+        },
+
+        /**
+         * Set event listener to track resize event
+         */
+        setEventListener: function () {
+            var running = false,
+                handler = function () {
+                    this.containerWidth = window.innerWidth;
+                    this.setLayoutStyles();
+                }.bind(this);
+
+            window.addEventListener('resize', function () {
+                if (!running) {
+                    running = true;
+
+                    if (window.requestAnimationFrame) {
+                        window.requestAnimationFrame(function () {
+                            handler();
+                            running = false;
+                        });
+                    } else {
+                        setTimeout(function () {
+                            handler();
+                            running = false;
+                        }, 66);
```

Comment:
Why 66?

Reply:
That is the optimal number for the timeout :) Any suggestions?

Reply:
1000 / 60

Reply:
^^you missed this comment

Reply:
Might make more sense to just use the polyfill/fallback we have in place [here](https://github.com/magento/magento2/blob/0eb8677b0b4e35606032e856cc1ef7c80e68829f/app/code/Magento/Ui/view/base/web/js/lib/view/utils/raf.js). At least that way we're not inlining these fallbacks everytime we need to use rAF

Reply:
I completely forgot that we have this.

Reply:
@DrewML @omiroshnichenko can you please clarify what would you like to see here

Reply:
Added raf